core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -431,6 +431,11 @@ String partitionsToLogString(Collection<TopicIdPartition> partitions) {
         return FetchSession.partitionsToLogString(partitions, log.isTraceEnabled());
     }
 
+    @Override
+    public void close() throws Exception {
```

Comment:
I don't see this method is being called from anywhere to close the timer object. You would require change in BrokerServer.scala

```
if (sharePartitionManager != null)
        CoreUtils.swallow(sharePartitionManager.close(), this)